sdk/src/client/secret/mod.rs

```diff
@@ -66,613 +57,504 @@ use crate::{
     },
 };
 
-/// The secret manager interface.
-#[async_trait]
-pub trait SecretManage: Send + Sync {
-    type Error: std::error::Error + Send + Sync;
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
+#[serde(rename_all = "camelCase")]
+pub struct PublicKeyOptions {
```

Comment:
What's the reason again to have this instead of Bip44 directly?

Reply:
`Bip44` is not our type, but we could also make it a `PublicKeyOptions(Bip44)`?

Reply:
I've thought about changing it, but of course then there's the mismatch between generating a single key and multiple

Reply:
Okay, but should we maybe name it `Bip44PublicKeyOptions` to make its purpose clearer?

Reply:
It's not very important, so you can resolve whether you do or not